src/engine/build.bat

```diff
@@ -106,19 +106,19 @@ if NOT "_%VS150COMNTOOLS%_" == "__" (
     set "BOOST_JAM_TOOLSET_ROOT=%VS150COMNTOOLS%..\..\VC\"
     goto :eof)
 call :Clear_Error
-if EXIST "%ProgramFiles%\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build\vcvarsall.bat"  (
+if EXIST "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build\vcvarsall.bat"  (
```

Comment:
> Is that not correct? I haven't had a 32-bit machine for a long time, so I can't easily look.

I don't have it too, but from my memory, old 32-bit Windows had only %ProgramFiles%. But I think it fine for now.

> If the VS15COMNTOOLS variable isn't reliable, then maybe we can query the registry or something to make it work?

It is not easy to do for VS2017 from scripting languages http://stackoverflow.com/q/41106407/61505